modules/core/src/main/java/org/apache/synapse/endpoints/Template.java

```diff
@@ -51,6 +51,11 @@
 
     private boolean isEdited;
 
+    /**
+     * Holds the list of comments associated with the proxy service.
+     */
+    private List<String> commentsList = new ArrayList<String>();
```

Comment:
Can we define this in the class SynapseArtifact, so that every artifacts class couls inherit it?

Reply:
There are certain classes such as SynapseConfiguration, Entry, ScheduledMessageForwardingProcessor, StartUpController etc that we don't require to persist the comments in. Therefore, is it necessary to add it to this layer?

Reply:
Since this is only being used in classes, it's ok to have the code in the child classes.